### ISSUE TITLE
Develop the react authentication form component

### ISSUE THREAD

**Issue by stynan27** [stynan27/BlockHeads#17]
### Description
Create a component that when clicking the appropriate button displays either a login form (login button) or a registration form ("Start Building!" button). This form will hook up to a Spring Boot API to be created this spring for creating and getting users' details.

- [x] Unblocked by completing #16 
- [x] and completing #20
~- [ ] and completing #18~ -> can be worked on separately for initial implementation (to be combined later in a separate task)
~- [ ] and completing #19~ -> moved to icebox

subtask of #10

**Comment by stynan27** [comment]
- [ ] Need to review Axios, get some examples for @jtangelo98

**Comment by stynan27** [comment]
The modal will display differently (Register vs Login) based on the button clicked to display this widget (i.e. "Login" vs "Start Building" buttons).

**Comment by stynan27** [comment]
At the end of the last sprint, we were running into state management issues related to the modal being present/not. I worked with @jtangelo98 to integrate the Redux package for centralized App state management. There are a few wrinkles to iron out still which Joey will be refining solo.
